src/software/geom/util.cpp

```diff
@@ -1000,16 +998,16 @@ Vector segmentNearLine(Vector a0, Vector a1, Vector b0, Vector b1)
     return p;
 }
 
-Vector intersection(Vector a1, Vector a2, Vector b1, Vector b2)
+Point intersection(Point a1, Point a2, Point b1, Point b2)
 {
     Vector a = a2 - a1;
 
     Vector b1r = (b1 - a1).rotate(-a.orientation());
     Vector b2r = (b2 - a1).rotate(-a.orientation());
     Vector br  = (b1r - b2r);
 
-    return Vector(b2r.x() - b2r.y() * (br.x() / br.y()), 0.0).rotate(a.orientation()) +
-           a1;
+    return a1 +
+           Vector(b2r.x() - b2r.y() * (br.x() / br.y()), 0.0).rotate(a.orientation());
```

Comment:
Do we only allow one ordering of the `+` operator?

Reply:
Yeah I figured only adding a Vector to a Point makes sense rather than adding a Point to a Vector

Reply:
Hmm, I'd be inclined to have both, just because addition is usually commutative (order doesn't matter). @MathewMacDougall - second thoughts on this?

Reply:
Hmmm that's a tough call. I think it's more readable if it's _not_ commutative, and you can only add a vector to a point, but technically Gareth is right about addition generally being commutative. I think the main question is only allowing one option more readable and worth the slight extra overhead writing the code?

@jonathanlew @EvanMorcom for their opinions.

Reply:
I think either Vector + Point and Point + Vector make sense intuitively. 

You can have a vector and want to shift it by a Point, or a Point and want a Vector towards something.

Reply:
Thanks for the review - added commutative addition for Points and Vectors in latest update.